gpytorch/kernels/kernel.py

```diff
@@ -456,10 +456,19 @@ def __init__(self, *kernels):
         self.kernels = ModuleList(kernels)
 
     def forward(self, x1, x2, **params):
-        res = lazify(self.kernels[0](x1, x2, **params))
+        x1_eq_x2 = torch.equal(x1, x2)
+
+        if not x1_eq_x2:
```

Comment:
Can you add comments here for why we're doing this? (Future Geoff might get stupid otherwise)

Reply:
Another way to do this: we could overwrite `LazyTensor.__mul__` to delazify if the matrix is not square/symmetric.

Reply:
^^ okay - we probably don't want to add that comment until we have flags that matrices are symmetric (because it's possible that the matrix is square but not symmetric). We probably do want to add these flags to LazyTensors eventually though.

Reply:
@gpleiss All set!